watertap/core/zero_order_costing.py

```diff
@@ -939,6 +940,49 @@ def cost_storage_tank(blk):
         ZeroOrderCostingData._general_power_law_form(
             blk, A, B, sizing_term, factor)
 
+    def cost_surface_discharge(blk):
+        """
+        General method for costing surface discharge. Capital cost is based on
+        construction and pipe costs.
+        """
+
+        t0 = blk.flowsheet().time.first()
+
+        # Get parameter dict from database
+        parameter_dict = \
+            blk.unit_model.config.database.get_unit_operation_parameters(
+                blk.unit_model._tech_type,
+                subtype=blk.unit_model.config.process_subtype)
+
+        # Get costing parameter sub-block for this technology
+        A, B, pipe_cost_basis, ref_state = _get_tech_parameters(
+            blk,
+            parameter_dict,
+            blk.unit_model.config.process_subtype,
+            ["capital_a_parameter", "capital_b_parameter", "pipe_cost_basis", "reference_state"])
+
+        # Add cost variable and constraint
+        blk.capital_cost = pyo.Var(
+            initialize=1,
+            units=blk.config.flowsheet_costing_block.base_currency,
+            bounds=(0, None),
+            doc="Capital cost of unit operation")
+
+        Q = pyo.units.convert(
+            blk.unit_model.properties[t0].flow_vol,
+            to_units=pyo.units.m ** 3 / pyo.units.hour)
+        expr = pyo.units.convert(
+            A * (Q / ref_state) ** B
```

Comment:
In this case, what you really want ot do is `pyo.units.convert(flow_vol/ref_state, to_units=pyo.units.dimensionless)`.

This allows for future cases where the units of `ref_state` are not known.

Reply:
Yes, you can nest convert statements as needed. So:

```
expr = pyo.units.convert(
            A * pyo.units.convert(Q / ref_state, to_units=pyo.units.dimensionless)**B,
            to_units=base_currency)
```

Reply:
I addressed this on my most recent update, so I am marking resolved here.